src/hash_to_curve.rs

```diff
@@ -128,7 +131,7 @@ where
     // 14. gx1 = gx1 + B
     let gx1 = gx1 + b;
     // 15. e1 = is_square(gx1)
-    let e1 = gx1.sqrt().is_some();
+    let e1 = !gx1.ct_quadratic_non_residue();
```

Comment:
why not `let e1 = gx1.ct_quadratic_residue();`

Reply:
`.sqrt().is_some()` will be true when called in `F::ZERO`  I wanted to ensure that `e1` was also true in that case. However, I'm not sure that `gx1 = 0` is a valid possibility.

Reply:
Ah I see, you are right, just checked the spec, `is_square` returns `true` when `gx1 = 0`